src/extension.ts

```diff
@@ -89,52 +95,30 @@ export async function activate(context: vscode.ExtensionContext) {
         })
         await ext.telemetry.start()
 
-        registerCommand({
-            command: 'aws.login',
-            callback: async () => await ext.awsContextCommands.onCommandLogin(),
-            telemetryName: 'aws_credentialslogin'
-        })
-
-        registerCommand({
-            command: 'aws.credential.profile.create',
-            callback: async () => await ext.awsContextCommands.onCommandCreateCredentialsProfile(),
-            telemetryName: 'aws_credentialscreate'
-        })
+        vscode.commands.registerCommand('aws.login', async () => await ext.awsContextCommands.onCommandLogin())
+        vscode.commands.registerCommand('aws.logout', async () => await ext.awsContextCommands.onCommandLogout())
 
-        registerCommand({
-            command: 'aws.logout',
-            callback: async () => await ext.awsContextCommands.onCommandLogout(),
-            telemetryName: 'aws_credentialslogout'
+        vscode.commands.registerCommand('aws.credential.profile.create', async () => {
+            await ext.awsContextCommands.onCommandCreateCredentialsProfile()
+            recordAwsCreateCredentials()
         })
 
         // register URLs in extension menu
-        registerCommand({
-            command: 'aws.help',
-            callback: async () => {
-                vscode.env.openExternal(vscode.Uri.parse(documentationUrl))
-            },
-            telemetryName: 'Command_aws.help'
+        vscode.commands.registerCommand('aws.help', async () => {
+            vscode.env.openExternal(vscode.Uri.parse(documentationUrl))
+            recordAwsHelp()
         })
-        registerCommand({
-            command: 'aws.github',
-            callback: async () => {
-                vscode.env.openExternal(vscode.Uri.parse(githubUrl))
-            },
-            telemetryName: 'Command_aws.github'
+        vscode.commands.registerCommand('aws.github', async () => {
+            vscode.env.openExternal(vscode.Uri.parse(githubUrl))
+            recordAwsShowExtensionSource()
         })
-        registerCommand({
-            command: 'aws.reportIssue',
-            callback: async () => {
-                vscode.env.openExternal(vscode.Uri.parse(reportIssueUrl))
-            },
-            telemetryName: 'Command_aws.reportIssue'
+        vscode.commands.registerCommand('aws.reportIssue', async () => {
+            vscode.env.openExternal(vscode.Uri.parse(reportIssueUrl))
+            recordAwsReportPluginIssue()
         })
-        registerCommand({
-            command: 'aws.quickStart',
-            callback: async () => {
-                await showQuickStartWebview(context)
-            },
-            telemetryName: 'Command_aws.quickStart'
+        vscode.commands.registerCommand('aws.quickStart', async () => {
+            await showQuickStartWebview(context)
```

Comment:
Metrics calls would be missed if function throws an Error

Reply:
fixed